### ISSUE TITLE
Latest gcloud sdk 349.0.0 breaks kpt installed manifests

### ISSUE THREAD

**Issue by ilkelma** [kubeflow/gcp-blueprints#294]
While running through the installation guide I encountered a blocker that required me to specifically downgrade the Google Cloud SDK installed version of `kpt` because the current manifests do not have the correct API version (`v1alpha1` instead of `v1`) for the latest version of `kpt` - `1.0.0-beta1`.

Specifically I was running through the guide in cloud shell as I have a Mac as my dev machine and the guide said bash would be an issue there.

I hit the following error (after the config connector stuff which i've omitted for brevity):

```
....
Build directory: ./build
Component path: common/asm
Apply component resources: common/asm
Found Makefile, call 'make apply' of this component Makefile.
make[1]: Entering directory '/home/aknight/gcp-blueprints/kubeflow/common/asm'
curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_1.9.3-asm.2-config2 > install_asm;
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 79280  100 79280    0     0  7038k      0 --:--:-- --:--:-- --:--:-- 7742k
rm -rf ./asm
kpt pkg get https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git/asm@1.9.3-asm.2+config2 ./
Package "asm":
Fetching https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages@1.9.3-asm.2+config2
From https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages
 * tag               1.9.3-asm.2+config2 -> FETCH_HEAD
Error: Kptfile at "https:/github.com/GoogleCloudPlatform/anthos-service-mesh-packages/asm@1.9.3-asm.2+config2" has an old version ("v1alpha1") of the Kptfile schema.
Please update the package to the latest format by following https://kpt.dev/installation/migration.
make[1]: *** [Makefile:43: download-asm-package] Error 1
make[1]: Leaving directory '/home/aknight/gcp-blueprints/kubeflow/common/asm'
make: *** [Makefile:67: apply] Error 1
```

When I run `kpt version` I see: `1.0.0-beta.1` which in the cloud shell case is installed by the apt package: 
```
Package: google-cloud-sdk-kpt
Version: 349.0.0-0
```

I looked at the `kpt` release notes and found this entry: https://github.com/GoogleContainerTools/kpt/releases/tag/v1.0.0-beta.1 which would mean the manifests need to be upgraded to `v1`

I was able to work around this by forcing a downgrade on the gcloud sdk installed version of `kpt` in cloud shell via this command: `sudo apt-get install google-cloud-sdk-kpt=346.0.0-0` which is prior to this version in which gcloud sdk upgraded the bundled version of `kpt`: https://cloud.google.com/sdk/docs/release-notes#34900_2021-07-20

With that workaround in place it installed just fine but as it stands the quickstart does not work with the latest `kpt` in general and also does not work with the latest `kpt` installed by the Google Cloud SDK.

**Comment by Bobgy** [comment]
Thanks for reporting the issue @ilkelma!

/cc @zijianjoy 
we will be supporting kpt v1 starting from Kubeflow 1.4.

Downgrading gcloud or installing a separate kpt binary https://github.com/GoogleContainerTools/kpt/releases/tag/v0.39.2 is current workaround.

Welcome contributions on doc updates or integrating kpt v1 with Kubeflow 1.3 if you are interested

**Comment by zijianjoy** [comment]
This issue should have been resolved with the latest gcloud version and kpt version (kpt v1.0.0-beta.6 or above). Would you like to give it a try? Thank you!

**Comment by zijianjoy** [comment]
Marking this issue as resolved as we have released Kubeflow v1.4 for a while now. Feel free to reopen this issue if problem still persists.
